### PR TITLE
[MM-9606] Fix cropping of single image preview of very wide or tall image

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix cropping of single image preview of very wide or tall image

#### Ticket Link
Jira ticket: [MM-9606](https://mattermost.atlassian.net/browse/MM-9606)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Has UI changes

### CODE REVIEW

File: components/single_image_view.jsx [hunk 1]
```diff
@@ -111,14 +110,14 @@ export default class SingleImageView extends React.PureComponent {
...
-        let previewWidth = fileInfo.width * 100;
-        let previewHeight = fileInfo.height * 100;
+        let previewWidth = fileInfo.width;
+        let previewHeight = fileInfo.height;
 
         if (viewPortWidth && previewWidth > viewPortWidth) {
             const origRatio = fileInfo.height / fileInfo.width;
-            previewWidth = Math.min(PREVIEW_IMAGE_MAX_WIDTH, fileInfo.width * 100, viewPortWidth);
+            previewWidth = Math.min(PREVIEW_IMAGE_MAX_WIDTH, fileInfo.width, viewPortWidth);
             previewHeight = previewWidth * origRatio;
```
Comment (saturninoabril) [inline components/single_image_view.jsx]: Not related - just reverted multiplying to 100.

File: components/single_image_view.jsx [hunk 2]
```diff
@@ -170,10 +168,22 @@ export default class SingleImageView extends React.PureComponent {
...
+            height = PREVIEW_IMAGE_MIN_DIMENSION;
+        }
+
+        let width = previewWidth;
+        if (width < PREVIEW_IMAGE_MIN_DIMENSION) {
+            width = PREVIEW_IMAGE_MIN_DIMENSION;
+        }
+
+        let imageStyle = {height};
+        const imageContainerStyle = {width};
```
Comment (saturninoabril) [inline components/single_image_view.jsx]: Set static width to `image-container` to properly scale the image based on height (or max-height). Currently, it's not scaling (whenever the viewport width is greater than the image width) but only maintained the ratio and then just hide the lower portion of the image.

File: sass/components/_files.scss [hunk 3]
```diff
@@ -168,7 +168,6 @@
         .image-loaded {
             @include single-transition(all, .1s, linear);
             @include border-radius(4px);
-            border: 1px solid transparent;
```
Comment (saturninoabril) [inline sass/components/_files.scss]: Deleted this border since it makes a slight scroll pop when applied.

File: components/single_image_view.jsx [hunk 4]
```diff
@@ -170,10 +171,25 @@ export default class SingleImageView extends React.PureComponent {
...
+        if (width < PREVIEW_IMAGE_MIN_DIMENSION) {
+            width = PREVIEW_IMAGE_MIN_DIMENSION;
+        }
+
+        let imageStyle = {height};
+        let imageLoadedStyle = {height};
+        let imageContainerStyle = {};
+        if (width < viewPortWidth && height === PREVIEW_IMAGE_MAX_HEIGHT) {
+            imageContainerStyle = {width};
+        }
```
Comment (saturninoabril) [inline components/single_image_view.jsx]: Only apply static width to image-container whenever scaling is necessary.

**Comment by saturninoabril** [comment]
@asaadmahmood The resizing on RHS toggle should work now. Static width for image-container is now only applied whenever scaling is necessary.
@esethna This should be ready for PM review. Please let me know if you find an issue.

**Comment by saturninoabril** [comment]
That image squish/distortion should be fixed by the latest commit (removing `min-height`). There maybe cache issue somewhere (browser/spinmit).
Spinmint is now working for me.

File: sass/components/_files.scss (hunk 3 above)
Comment (esethna) [reply sass/components/_files.scss]: Sorry, we can't really delete this. Is the scroll pop caused by a 1px border really that noticable?

**Comment by saturninoabril** [comment]
@esethna Yes, I think roughly 3-5px, it varies. I suggest not to have it at the moment.

File: sass/components/_files.scss (hunk 3 above)
Comment (saturninoabril) [reply sass/components/_files.scss]: Yes, I think roughly 3-5px, it varies and it's not consistent depending on whether the image is at max-height of 350px. I suggest not to have it at the moment.

**Comment by esethna** [comment]
Also, is it intended that the image doesn't scale when the RHS is opened (instead it crops). Is this to reduce scroll pop?

**Comment by saturninoabril** [comment]
There are cases it will not scale like when it initially render below 50px. With that, it will crop the image only.

**Pull request merged**